models/src/main/scala/coop/rchain/models/SortedParMap.scala

```diff
@@ -1,29 +1,50 @@
 package coop.rchain.models
 
-import scala.collection.MapLike
 import coop.rchain.models.rholang.sorter.ordering._
 
-class SortedParMap private (private val ps: Map[Par, Par])
-    extends Map[Par, Par]
-    with MapLike[Par, Par, SortedParMap] {
+import scala.collection.GenTraversableOnce
+import scala.collection.immutable.ListMap
 
-  lazy val sortedMap: Map[Par, Par] = ps.sort.toMap
+final class SortedParMap private (ps: Map[Par, Par]) extends Iterable[(Par, Par)] {
```

Comment:
Why do you say we should implement IterableLike instead of Iterable?

Reply:
Not instead but in addition to. However seems like it is not needed for our purposes.